drivers/adc/adc_mcux_lpadc.c

```diff
@@ -403,15 +402,26 @@ static int mcux_lpadc_init(const struct device *dev)
 	struct mcux_lpadc_data *data = dev->data;
 	ADC_Type *base = config->base;
 	lpadc_config_t adc_config;
-#ifdef CONFIG_PINCTRL
 	int err;
 
+#ifdef CONFIG_PINCTRL
 	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
 	if (err) {
 		return err;
 	}
 #endif
 
+	/* Enable necessary regulators */
+	const struct device **regulator = config->req_regulators;
+
+	while (*regulator != NULL) {
+		err = regulator_enable(*regulator);
+		regulator++;
+		if (err) {
+			return err;
+		}
+	}
```

Comment:
shouldn't this be enabled only when needed?

Reply:
note here as well: driver should `select REGULATOR`.

Reply:
> shouldn't this be enabled only when needed?

I am not clear what you mean by this